### DISCUSSION TITLE
Sidebar scroll to active menu

### DISCUSSION THREAD

**Discussion by gulgil** [serenity-is/Serenity#6789] [Help]
We have a very large sidebar menu, and even though we have grouped what we can it's not quite enough. Some active menus will be off screen when we click them and the page reloads. Is there anyway to get the menu to scroll the active menuitem into view when it loads?

The menu have "active" as class, but so does all it's parents. I'm thinking a search for the deepest element with "active" class, and then a scrollTo() call. 

Any help on how to do this?

**Comment by wezmag** [comment]
1. To get the top of the active element
``` js
const top = document.querySelector(".s-sidebar-item.active").offsetTop;
```
2. Scroll to the element
``` js
document.querySelector("s-sidebar-pane").scrollTo(0, top);
```
3. put together in the `common-theme.js` (Serene)
``` js
document.addEventListener('DOMContentLoaded', function () {
    //...
    const activeItem = document.querySelector(".s-sidebar-item.active");
    if (activeItem) {
        document.querySelector(".s-sidebar-pane").scrollTo(0, activeItem.offsetTop);
    }
});
```
